### ISSUE TITLE
buildbot tests failing on 18.09

### ISSUE THREAD

**Issue by livnev** [NixOS/nixpkgs#56053]
## Issue description

The tests for `buildbot` seem to be failing. Full test error output is [here](https://gist.github.com/livnev/37d9566b8721afabeb04dd92962d511d). Modifying `doCheck = false;` in the derivation for `buildbot` works as a band aid. However, the package is written in such a way that I couldn't figure out how to override the `doCheck` attribute, so my only option seemed to be to use a custom `nixpkgs` :disappointed: 

### Steps to reproduce

Try to install `buildbot-full`

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.13, NixOS, 18.09.2203.9bd45dddf81 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(lev): `""`
 - channels(root): `"nixos-18.09.2203.9bd45dddf81"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by veprbl** [comment]
It seems like `buildbot` was never successfully built on release 18.09 branch: https://hydra.nixos.org/job/nixos/release-18.09/nixpkgs.buildbot.x86_64-linux/all?page=1

Initially (at 18.09 tag) it didn't evaluate:
```
error: pylint-2.1.1 not supported for interpreter python2.7
```
Then `pylint` was backported in 0da77aaaf82, so build of `buildbot` started to be executed, but it was never fixed after that.

cc @nand0p @ryansydnor

**Comment by lopsided98** [comment]
Backporting https://github.com/NixOS/nixpkgs/commit/52307dff2572788e79391d766158a5cda52e87cb should fix the test failures.

**Comment by veprbl** [comment]
@lopsided98 Thanks for the tip!

**Comment by veprbl** [comment]
Fixed on release-18.09 now: https://hydra.nixos.org/eval/1508301#tabs-now-succeed
